x/httpconnect/transport.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2025 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpconnect
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+	"github.com/quic-go/quic-go"
+	"github.com/quic-go/quic-go/http3"
+	"golang.org/x/net/http2"
+)
+
+type TransportOption func(c *transportConfig)
+
+// WithTLS configures the transport to use TLS.
+func WithTLS(tlsConf *tls.Config) TransportOption {
```

Comment:
Please use the config in our tls package instead. Go's TLS Config is hard to reason about.